python/nrel/routee/compass/compass_app.py

```diff
@@ -73,7 +77,7 @@ def from_dict(cls, config: Dict, working_dir: Optional[Path] = None) -> CompassA
         """
         path_str = str(working_dir.absolute()) if working_dir is not None else ""
         toml_string = toml.dumps(config)
-        app = CompassAppWrapper._from_config_toml_string(toml_string, path_str)
+        app = cls.get_constructor()._from_config_toml_string(toml_string, path_str)
```

Comment:
we don't need get_constructor, that's what `cls` is, you could just call `cls._from_config_toml_string` directly.

Reply:
or, am i getting this wrong, because it's coming from a macro?